x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/full_time_range_selector/full_time_range_selector.tsx

```diff
@@ -54,19 +77,113 @@ export const FullTimeRangeSelector: FC<Props> = ({
       );
     }
   }
-  return (
-    <EuiButton
-      isDisabled={disabled}
-      onClick={() => setRange(indexPattern, query)}
-      data-test-subj="dataVisualizerButtonUseFullData"
-    >
-      <FormattedMessage
-        id="xpack.dataVisualizer.index.fullTimeRangeSelector.useFullDataButtonLabel"
-        defaultMessage="Use full {indexPatternTitle} data"
-        values={{
-          indexPatternTitle: indexPattern.title,
+
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const [frozenDataPreference, setFrozenDataPreference] = useStorage<FrozenTierPreference>(
+    DV_FROZEN_TIER_PREFERENCE,
+    // By default we will exclude frozen data tier
+    FROZEN_TIER_PREFERENCE.EXCLUDE
+  );
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const sortOptions: EuiRadioGroupOption[] = useMemo(() => {
+    return [
+      {
+        id: FROZEN_TIER_PREFERENCE.EXCLUDE,
+        label: i18n.translate(
+          'xpack.dataVisualizer.index.fullTimeRangeSelector.useFullDataExcludingFrozenMenuLabel',
+          {
+            defaultMessage: 'Exclude frozen data tier',
+          }
+        ),
+      },
+      {
+        id: FROZEN_TIER_PREFERENCE.INCLUDE,
+        label: i18n.translate(
+          'xpack.dataVisualizer.index.fullTimeRangeSelector.useFullDataIncludingFrozenMenuLabel',
+          {
+            defaultMessage: 'Include frozen data tier',
+          }
+        ),
+      },
+    ];
+  }, []);
+
+  const popoverContent = (
+    <EuiPanel>
+      <EuiRadioGroup
+        options={sortOptions}
+        idSelected={frozenDataPreference}
+        onChange={(id) => {
+          setFrozenDataPreference(id as FrozenTierPreference);
+          setRange(indexPattern, query, id === FROZEN_TIER_PREFERENCE.EXCLUDE);
+          closePopover();
         }}
+        compressed
       />
-    </EuiButton>
+    </EuiPanel>
+  );
+
+  const buttonTooltip =
+    frozenDataPreference === FROZEN_TIER_PREFERENCE.EXCLUDE ? (
+      <FormattedMessage
+        id="xpack.ml.fullTimeRangeSelector.useFullDataButtonTooltip"
+        defaultMessage="Use full range of data excluding frozen data tier."
+      />
+    ) : (
+      <FormattedMessage
+        id="xpack.ml.fullTimeRangeSelector.useFullDataButtonTooltip"
+        defaultMessage="Use full range of data including frozen data tier. Searches with frozen data may take longer."
```

Comment:
IMO you can merge these sentences. For example:
```suggestion
        defaultMessage="Use full range of data including frozen data tier, which might have slower search results."
```

Reply:
Updated [here](https://github.com/elastic/kibana/pull/122306/commits/a91b6685d2033576162098e0d879bd4a97f07144) and [here](https://github.com/elastic/kibana/pull/122306/commits/efbb43a31b7432a63ad3ff1e00e375655a9676cc)